clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -1229,20 +1253,76 @@ check_false_syscalls()
                          "Failed to catch misplaced syscall marker")) {
             res = false;
         }
     }
+    // Ensure timestamps are where we expect them.
+    std::cerr << "Testing syscall timestamps\n";
+    {
+        // Correct: syscall preceded by timestamp+cpu.
+        std::vector<memref_with_IR_t> memref_setup = {
+            { gen_marker(TID, TRACE_MARKER_TYPE_VERSION,
+                         TRACE_ENTRY_VERSION_ACCURATE_TIMESTAMPS),
+              nullptr },
+            { gen_marker(TID, TRACE_MARKER_TYPE_FILETYPE, FILE_TYPE), nullptr },
+            { gen_instr(TID), sys },
+            { gen_marker(TID, TRACE_MARKER_TYPE_TIMESTAMP, 101), nullptr },
+            { gen_marker(TID, TRACE_MARKER_TYPE_CPU_ID, 3), nullptr },
+            { gen_marker(TID, TRACE_MARKER_TYPE_SYSCALL, 42), nullptr },
+        };
+        auto memrefs = add_encodings_to_memrefs(ilist, memref_setup, BASE_ADDR);
+        if (!run_checker(memrefs, false))
+            res = false;
+    }
+    {
+        // Incorrect: syscall with no preceding timestamp+cpu.
+        std::vector<memref_with_IR_t> memref_setup = {
+            { gen_marker(TID, TRACE_MARKER_TYPE_VERSION,
+                         TRACE_ENTRY_VERSION_ACCURATE_TIMESTAMPS),
+              nullptr },
+            { gen_marker(TID, TRACE_MARKER_TYPE_FILETYPE, FILE_TYPE), nullptr },
+            { gen_instr(TID), sys },
+            { gen_marker(TID, TRACE_MARKER_TYPE_SYSCALL, 42), nullptr },
+        };
+        auto memrefs = add_encodings_to_memrefs(ilist, memref_setup, BASE_ADDR);
+        if (!run_checker(memrefs, true,
+                         { "Syscall marker not preceded by timestamp + cpuid", TID,
+                           /*ref_ordinal=*/4, /*last_timestamp=*/0,
+                           /*instrs_since_last_timestamp=*/1 },
+                         "Failed to catch syscall without timestamp+cpuid")) {
+            res = false;
+        }
+    }
+    {
+        // Incorrect: syscall with preceding cpu but no timestamp.
+        std::vector<memref_with_IR_t> memref_setup = {
+            { gen_marker(TID, TRACE_MARKER_TYPE_VERSION,
+                         TRACE_ENTRY_VERSION_ACCURATE_TIMESTAMPS),
+              nullptr },
+            { gen_marker(TID, TRACE_MARKER_TYPE_FILETYPE, FILE_TYPE), nullptr },
+            { gen_instr(TID), sys },
+            { gen_marker(TID, TRACE_MARKER_TYPE_CPU_ID, 3), nullptr },
+            { gen_marker(TID, TRACE_MARKER_TYPE_SYSCALL, 42), nullptr },
+        };
+        auto memrefs = add_encodings_to_memrefs(ilist, memref_setup, BASE_ADDR);
+        if (!run_checker(memrefs, true,
+                         { "Syscall marker not preceded by timestamp + cpuid", TID,
+                           /*ref_ordinal=*/5, /*last_timestamp=*/0,
+                           /*instrs_since_last_timestamp=*/1 },
+                         "Failed to catch syscall without timestamp")) {
+            res = false;
+        }
+    }
```

Comment:
Worth noting in a comment here that a missing test for post-syscall timestamp+cpu markers is not an oversight but intentionally skipped as syscalls without post-syscall event won't have that.

Reply:
Done.